### DISCUSSION TITLE
Tectonic Project Support

### DISCUSSION THREAD

**Discussion by dougransom** [James-Yu/LaTeX-Workshop#3107] [Q&A]
[Tectonic](https://tectonic-typesetting.github.io)-specific projects may have a [Tectonic.toml](https://tectonic-typesetting.github.io/book/latest/ref/tectonic-toml.html) file which specifies the project.

To build it once, we would use the command:

[`tectonic -X build`](https://tectonic-typesetting.github.io/book/latest/v2cli/build.html)

To build it whenever a file has changed in the project, we would use the command:
 

[`tectonic -X watch`](https://tectonic-typesetting.github.io/book/latest/v2cli/watch.html)

I would suggest the LaTex-Workshop have a way for the users to initiate either of those two commmoands in the folder where a  [Tectonic.toml](https://tectonic-typesetting.github.io/book/latest/ref/tectonic-toml.html) exists, perhaps witha build icon of some sort left of the [Tectonic.toml](https://tectonic-typesetting.github.io/book/latest/ref/tectonic-toml.html)  int the explorer.

**Comment by jlelong** [comment]
Supporting `tectonix -X watch` is out of the scope of this extension.

You can always define a recipe to run `tectonix -X build` to build your compile. See https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-recipes
